### PR TITLE
[develop] Updated modulefiles for Cheyenne, Hera, Jet, Gaea, Orion 

### PR DESCRIPTION
**Pull request by natalie-perlin**
## DESCRIPTION OF CHANGES: 

Updated modulefiles for the following HPC systems: **Cheyenne**, **Gaea**, **Hera**, **Orion**, **Jet**, in particular, for the SRW 2.1 release. The changes include new paths for HPC-stack and miniconda3 installations in EPIC-managed common space. Met and Metplus verification packages were installed as part of the hpc-stack. Rocoto has also been installed on Cheyenne in  common EPIC-manages space. Met and Metplus modules are not loaded explicitly as modules, but the installation paths are specified in ./ush/machine/<_platform_>.yaml files. The packages needed for plotting routines are included in the **regional_workflow** conda virtual environment.

The following types of modulefiles were updated:

./modulefiles/build_<_platform_>_<compiler>
./modulefiles/wflow_<_platform_>
./modulefiles/tasks/<_platform_>/miniconda_regional_workflow
./modulefiles/tasks/<_platform_>/make_grid.local, make_ics.local, make_lbcs.local, make_orog.local, get_extrn_lbcs.local, get_extrn_ics.local, run_fcst.local, run_vx.local
./ush/machine/<_platform_>.yaml

where <_platform_> is **Cheyenne**, **Jet**, **Hera**, **Orion**, **Gaea**, 
and compiler either <_intel_> or <_gnu_>; Cheyenne has both compilers

A list of the recent hpc-stack builds updated recently for the current PR (plus additional compiler-mpi combinations for Hera) could be found in the following open issue in the [UFS-WM/Issues-1465:](https://github.com/ufs-community/ufs-weather-model/issues/1465).

### Type of change

- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## TESTS CONDUCTED: 

- [x] hera.intel - building SRW, running tests via Jenkins, basic plots via Jenkins
- [x] orion.intel - building SRW, running tests via Jenkins, basic plots via Jenkins
- [x] cheyenne.intel - building SRW, running tests via Jenkins
- [x] cheyenne.gnu - building SRW on
- [x] gaea.intel  - building SRW, running tests via Jenkins, basic plots via Jenkins
- [x] jet.intel -  building SRW, running tests via Jenkins, basic plots via Jenkins
- [ ] wcoss2.intel
- [ ] NOAA Cloud (indicate which platform)
- [x] Jenkins
- [ ] fundamental test suite
- [ ] comprehensive tests (specify *which* if a subset was used)

## DEPENDENCIES:

Documentation needs to be updated, including the Chapter on making plots.

## CHECKLIST

- [x] My code follows the style guidelines in the Contributor's Guide
- [x] I have performed a self-review of my own code using the Code Reviewer's Guide
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [ ] My changes do not require updates to the documentation (explain).
- [ ] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

## LABELS (optional): 

A Code Manager needs to add the following labels to this PR: 
- [ ] Work In Progress
- [ ] bug
- [x] enhancement
- [ ] documentation
- [ ] release
- [ ] high priority
- [ ] run_ci
- [ ] run_we2e_fundamental_tests
- [ ] run_we2e_comprehensive_tests
- [ ] Needs Cheyenne test 
- [ ] Needs Jet test 
- [ ] Needs Hera test 
- [ ] Needs Orion test 
- [ ] help wanted

## CONTRIBUTORS (optional): 

@BruceKropp-Raytheon (Jenkinks jobs building the SRW and running the tests)
@EdwardSnyder-NOAA (tests on Gaea and Hera, met/metplus verification)

### CODE REVIEW

**Comment by danielabdi-noaa** [comment]
@natalie-perlin Do you know what version of Lmod is available on Cheyenne? We need something above version `8.3.7` for the Lua modulefiles.

**Comment by natalie-perlin** [comment]
@danielabdi-noaa  - Cheyenne system-wide install uses Lmod 8.1.7. If a higher version is needed, it could also be installed in EPIC-managed space, similarly to Gaea. Is it a preferred approach?

**Comment by danielabdi-noaa** [comment]
@natalie-perlin Yes, that would be useful since we need >8.3.7 for Lua modulefiles. We had a similar issue with python version but that is not a problem anymore. Thanks!

**Comment by natalie-perlin** [comment]
@danielabdi-noaa  - I've installed Lua 5.4.4 locally on Cheyenne, and starting to install Lmod/8.7.13. The system went down this morning, and I'll resume the installation when Cheyenne is back. Please feel free to comment if any of the versions might not be what's required.

**Comment by MichaelLueken** [comment]
@natalie-perlin Has a similar PR been opened in the ufs-weather-model repo?  We will need to coordinate this update with @jkbk2004 so that all EPIC repositories use the same modulefiles from the same location.  Also, do you or Bruce Kropp have the Jenkins pipeline URL for the tests that were run?

**Comment by natalie-perlin** [comment]
@MichaelLueken - OK, good to know about the need to open a similar PR for the ufs-weather-model. Will take care of it asap, and coordinate with @jkbk2004. And yes, Bruce Kropp had several Jenkins tests done, need to ask @BruceKropp-Raytheon about the links

**Comment by MichaelLueken** [comment]
@natalie-perlin Great!  Thanks for working on creating a similar PR for the ufs-weather-model.  When it comes to stacks and library updates, these should be made to the ufs-weather-model first, then once they have been validated and approved, then the SRW can move forward with the update.

@BruceKropp Do you have the Jenkins pipeline links for the tests that were run for this PR? I'd like to review them before we kick off the Jenkins fundamental tests for this PR.

**Comment by natalie-perlin** [comment]
@MichaelLueken  - some notes on modules for the ufs-weather-model vs for the ufs-srweather-app:
The major difference with the weather model is that it does not need any specific python or conda environment, and therefore does not need to have the HPC modules built in relation to a specific python or conda. SRW and the workflow on the other hand, depends greatly on the conda and its conda environment. Additionally, the hpc modules built for the SRW include packages for met and metplus verification, installed under the same hpc stack directory tree.    Some of the modules for the ufs-weather-model may also test higher-version package options, but which may not be necessary for the SRW.

**Comment by MichaelLueken** [comment]
@natalie-perlin Thanks for bringing this to my attention.  The issue is that we can't have two different EPIC maintained stacks for two different EPIC maintained repositories.  Both the ufs-srweather-app and ufs-weather-model need to use the same stack, which is an issue currently.

Do the modules in the hpc-stack require to be built differently for python/conda environment purposes?  Would modules built specifically for the regional_workflow conda environment not work for building the weather model?  If this is the case, then multiple versions for several modules will likely be required.  As an example, the stacks from issue #409 would need to be updated to include `bufr/11.7.0` (only `bufr/11.5.0` is available in these stacks), `ncio/1.1.2` (only `ncio/1.1.0` is available in these stacks), and `nccmp/1.8.9.0` (no version of `nccmp` is available in these stacks).  These would be required to be created in a way so that they can be used in the regional_workflow conda environment.

As new library versions are made available, the stacks will need to be updated to include these updated versions.  These versions will need to be tested and evaluated with the weather model, then they can be brought over to the SRW, tested, and used.

**Comment by natalie-perlin** [comment]
@MichaelLueken  - 
  All the modules you mentioned,` bufr/11.7.0`, `ncio/1.1.2`, and `nccmp/1.8.9.0` are built in the locations for  this R-419. I just have added nccmp/1.8.9.0 today to Gaea and Orion and set as a default version (hpc-stack had different version build earlier).

**Comment by MichaelLueken** [comment]
> @MichaelLueken - All the modules you mentioned,` bufr/11.7.0`, `ncio/1.1.2`, and `nccmp/1.8.9.0` are built in the locations for this R-419. I just have added nccmp/1.8.9.0 today to Gaea and Orion and set as a default version (hpc-stack had different version build earlier).

@natalie-perlin - Sorry about that, I meant the paths to the hpc-stacks on Gaea, Orion, and Jet, not the updated hpc-stacks that you have created for this PR:

```
Orion: /work/noaa/epic-ps/hpc-stack/libs/intel/2022.1.2/modulefiles/stack
Gaea: /lustre/f2/pdata/ncep_shared/hpc-stack.epic/libs/intel/2021.3.0/modulefiles/stack
Jet: /lfs4/HFIP/hfv3gfs/nwprod/hpc-stack.epic/libs/intel/2022.1.2/modulefiles/stack
```

These three stack locations would need to be updated before we can try and move forward with using them for the SRW.

**Comment by natalie-perlin** [comment]
@MichaelLueken - these stack locations were not built using `role-epic` account, but under` jongkim`/`Jong.Kim` username. There are no logs, or indication how they were configured and location of hpc-stack source code directories (only guessing), and thus only Jong Kim (@jkbk2004) could know the way they were built, and could update them.

**Comment by MichaelLueken** [comment]
@natalie-perlin Yes, you are correct that only Jong can update the modules in the paths that I noted above and in issue #409.  This was just pointing out that we need to sync the hpc-stack used for both SRW and the weather model.  It sounds like you might need to add some updated versions for the weather model to use the epic.role stack locations, while Jong would need to add several modules to his personal stack locations before the SRW would be able to use his personal stack locations.  Again, the primary goal is to use a single stack for both the SRW and weather model.  You have completed the necessary work to use the epic.role version for the SRW, but these locations will also need to work for the weather model, with a PR opened in the weather model repository updating the location of the stack.

**Comment by panll** [comment]
It was tested on Hera and works fine.

**Review by panll — APPROVED** [review]

**Comment by BruceKropp-Raytheon** [comment]
e2e tests not yet passing on Jet.

File: modulefiles/build_cheyenne_gnu [hunk 1]
```diff
@@ -8,17 +8,19 @@ proc ModulesHelp { } {
 module-whatis "Loads libraries needed for building SRW on Cheyenne"
 
 module load cmake/3.22.0
-module load python/3.7.9
 module load ncarenv/1.3
-module load gnu/11.2.0
+module load gnu/12.1.0
```
Comment (MichaelLueken) [inline modulefiles/build_cheyenne_gnu:12]: Why was the compiler version updated to the latest on Cheyenne?  Would it be possible to bring this back down to `gnu/11.2.0`?

File: modulefiles/build_hera_intel [hunk 2]
```diff
@@ -2,23 +2,25 @@
...
 
 module load cmake/3.20.1
 
-module use /scratch2/NCEPDEV/nwprod/hpc-stack/libs/hpc-stack/modulefiles/stack
+module use /scratch1/NCEPDEV/nems/role.epic/hpc-stack/libs/intel-2022.2.0/modulefiles/stack
 module load hpc/1.2.0
-module load hpc-intel/2022.1.2
-module load hpc-impi/2022.1.2
+module load hpc-intel/2022.2.0
+module load hpc-impi/2022.2.0
```
Comment (MichaelLueken) [inline modulefiles/build_hera_intel]: Why were the compiler and mpi versions updated to the latest on Hera? Would it be possible to bring these back down to `intel/2022.1.2` and `impi/2022.1.2`?

File: modulefiles/build_jet_intel [hunk 3]
```diff
@@ -2,21 +2,25 @@
...
+module load cmake/3.20.1
+module load intel/2022.2.0
+module load impi/2022.2.0
 
+module use /mnt/lfs4/HFIP/hfv3gfs/role.epic/hpc-stack/libs/intel-2022.2.0/modulefiles/stack
 module load hpc/1.2.0
-module load hpc-intel/2022.1.2
-module load hpc-impi/2022.1.2
+module load hpc-intel/2022.2.0
+module load hpc-impi/2022.2.0
```
Comment (MichaelLueken) [inline modulefiles/build_jet_intel]: Why were the compiler and mpi versions updated to the latest on Jet? Would it be possible to bring these back down to `intel/2022.1.2` and `impi/2022.1.2`?

**Review by MichaelLueken — COMMENTED** [review]
@natalie-perlin Would it be possible to roll the compiler and mpi versions back to what is currently being used on Cheyenne, Hera, and Jet?  While it is nice to see that the SRW app builds and runs the workflow using these updated compilers and mpi versions, none of our WE2E tests check for reproducibility, among other aspects.

As is, the ufs-weather-model regression tests will need to be run to see if these updated compilers and mpi versions affect results.  This lies outside the scope of the ufs-srweather-app and is one of the reason why a similar PR needs to opened for the ufs-weather-model repository (the other being to ensure that both the SRW app and weather model use the same EPIC maintained stacks).

File: modulefiles/build_jet_intel (hunk 3 above)
Comment (natalie-perlin) [reply modulefiles/build_jet_intel]: Sure, this is not a problem. Will prepare it for the earlier version of intel/2022.1.2. The idea was to update it to the most recent compiler/mpi combination

File: modulefiles/build_cheyenne_gnu (hunk 1 above)
Comment (natalie-perlin) [reply modulefiles/build_cheyenne_gnu:12]: The attempt was to use the latest available compiler. The same libraries for the gnu/11.2.0 will be built wehn cheyenne comes back from the maintenance, if this is preferred.

File: modulefiles/build_hera_intel (hunk 2 above)
Comment (natalie-perlin) [reply modulefiles/build_hera_intel]: The attempt was to use the latest available compiler. The same libraries for the intel/2021.1.2 are being built and the PR will be updates asap, if this intel/2021.1.2 is preferred (tested with the WM)

File: modulefiles/build_cheyenne_gnu (hunk 1 above)
Comment (MichaelLueken) [reply modulefiles/build_cheyenne_gnu:12]: Thanks @natalie-perlin.  Yes, please built the same libraries for `gnu/11.2.0` on Cheyenne once it is back up.

File: modulefiles/build_hera_intel (hunk 2 above)
Comment (MichaelLueken) [reply modulefiles/build_hera_intel]: Thanks @natalie-perlin!  Yes, please build the same libraries using `intel/2022.1.2` on Hera.

File: modulefiles/build_jet_intel (hunk 3 above)
Comment (MichaelLueken) [reply modulefiles/build_jet_intel]: Thanks @natalie-perlin!

**Comment by natalie-perlin** [comment]
The hpc libraries and modules built with intel/2022.1.2 compiler and impi/2022.1.2 have been prepared for Hera and Jet.

**Comment by BruceKropp-Raytheon** [comment]
e2e tests verified on Jet. thanks.

**Comment by natalie-perlin** [comment]
Modulefiles in SRW were updated to use EPIC-managed miniconda3/4.12.0 for the workflow and running tasks. The 
[PR#444](https://github.com/ufs-community/ufs-srweather-app/pull/444)  (@mark-a-potts) included these changes and has been already approved and merged into develop

**Comment by MichaelLueken** [comment]
@natalie-perlin With the merge of PR #549 (at [d036849](https://github.com/ufs-community/ufs-srweather-app/commit/d03684908562a7118f9c369852cdfe2b66371977)), the ufs-srweather-app is now using your EPIC-maintained HPC-Stack locations on all machines.  I am now closing this PR.

If you feel that the PR should be reopened, please let me know.

**Pull request closed without merging**